docs/design/architecture.md

```diff
@@ -0,0 +1,89 @@
+---
+title: Architecture
+sidebar_position: 1
+---
+
+Quickwit is built on three pillars:
```

Comment:
```suggestion
Search in Quickwit  is built on three pillars:
```
without the search part this list feels a little incoherent

Reply:
yes this needs a complete rewrite. I just moved the files for now and everything else is on notion